tests/test_filecache.py

```diff
@@ -806,37 +1063,119 @@ def test_local_upl_pfx_bad():
-def test_cloud_upl_bad(prefix):
+def test_cloud_upl_bad_1(prefix):
     with FileCache() as fc:
         with pytest.raises(FileNotFoundError):
             fc.upload(f'{prefix}/{uuid.uuid4()}/XXXXXXXXX.xxx',
                       anonymous=True)
+        assert fc.download_counter == 0
+        assert fc.upload_counter == 0
+        ret = fc.upload(f'{prefix}/{uuid.uuid4()}/XXXXXXXXX.xxx',
+                        anonymous=True, exception_on_fail=False)
+        assert isinstance(ret, FileNotFoundError)
+        assert fc.download_counter == 0
+        assert fc.upload_counter == 0
     assert not fc.cache_dir.exists()
 
 
 @pytest.mark.parametrize('prefix', WRITABLE_CLOUD_PREFIXES)
-def test_cloud_upl_pfx_bad(prefix):
+def test_cloud_upl_pfx_bad_1(prefix):
     with FileCache() as fc:
         new_prefix = f'{prefix}/{uuid.uuid4()}'
         pfx = fc.new_prefix(new_prefix, anonymous=True)
         with pytest.raises(FileNotFoundError):
             pfx.upload('XXXXXXXXX.xxx')
+        assert fc.download_counter == 0
+        assert fc.upload_counter == 0
+        assert pfx.download_counter == 0
+        assert pfx.upload_counter == 0
+        ret = pfx.upload('XXXXXXXXX.xxx', exception_on_fail=False)
+        assert isinstance(ret, FileNotFoundError)
+        assert fc.download_counter == 0
+        assert fc.upload_counter == 0
+        assert pfx.download_counter == 0
+        assert pfx.upload_counter == 0
+    assert not fc.cache_dir.exists()
+
+
+@pytest.mark.parametrize('prefix', BAD_WRITABLE_CLOUD_PREFIXES)
+def test_cloud_upl_bad_2(prefix):
+    with FileCache() as fc:
+        new_prefix = f'{prefix}/{uuid.uuid4()}'
+        filename = 'test_file.txt'
+        local_path = fc.get_local_path(f'{new_prefix}/{filename}')
+        _copy_file(EXPECTED_DIR / EXPECTED_FILENAMES[0], local_path)
+        with pytest.raises(Exception) as e:
+            fc.upload(f'{new_prefix}/{filename}', anonymous=True)
+        assert not isinstance(e.type, FileNotFoundError)
+        assert fc.download_counter == 0
+        assert fc.upload_counter == 0
+        ret = fc.upload(f'{new_prefix}/{filename}', anonymous=True,
+                        exception_on_fail=False)
+        assert isinstance(ret, Exception)
+        assert not isinstance(e, FileNotFoundError)
+        assert fc.download_counter == 0
+        assert fc.upload_counter == 0
+    assert not fc.cache_dir.exists()
+
+
+@pytest.mark.parametrize('prefix', BAD_WRITABLE_CLOUD_PREFIXES)
+def test_cloud_upl_pfx_bad_2(prefix):
+    with FileCache() as fc:
+        new_prefix = f'{prefix}/{uuid.uuid4()}'
+        pfx = fc.new_prefix(new_prefix, anonymous=True)
+        filename = 'test_file.txt'
+        local_path = pfx.get_local_path(filename)
+        _copy_file(EXPECTED_DIR / EXPECTED_FILENAMES[0], local_path)
+        with pytest.raises(Exception) as e:
+            pfx.upload(filename)
+        assert not isinstance(e.type, FileNotFoundError)
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/SETI/rms-filecache/pull/9#issuecomment-2423431708>#25082a</a><span> - 10/19/2024, 01:42 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The change from 'e' to 'e.type' is incorrect. When using pytest.raises, the exception object should be accessed via the 'value' attribute, not 'type'. A similar issue was also found in tests/test_filecache.py (line 1144).
</div>
<div id="fix">
<b>Fix:</b> Change 'e.type' back to 'e.value' to correctly access the exception object in the pytest.raises context.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1166 +1166 @@
 -        assert not isinstance(e.type, FileNotFoundError)
 +        assert not isinstance(e.value, FileNotFoundError)
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged